src/View/Widget/DateTimeWidget.php

```diff
@@ -218,11 +218,11 @@ protected function _deconstructDate($value, $options)
             ];
         }
         try {
-            if (is_string($value)) {
+            if (is_string($value) && !is_numeric($value)) {
                 $date = new DateTime($value);
             } elseif (is_bool($value)) {
                 $date = new DateTime();
-            } elseif (is_int($value)) {
+            } elseif (is_int($value) || is_numeric($value)) {
                 $date = new DateTime('@' . $value);
```

Comment:
I'm not sure we can easily accept timestamps in arbitrary timezones. Taking in timezone parameters gets a bit complicated when dealing with POST data. Having string timestamps work like integer timestamps (which they didn't before) seems like a good idea to me.